types/ref-array-di/index.d.ts

```diff
@@ -0,0 +1,62 @@
+// Type definitions for ref-array-di 1.2
+// Project: https://github.com/node-ffi-napi/ref-array-di
+// Definitions by: Keerthi Niranjan <https://github.com/keerthi16>, Kiran Niranjan <https://github.com/KiranNiranjan>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import ref = require('ref-napi');
+
+interface ArrayType<T> extends ref.Type {
+    BYTES_PER_ELEMENT: number;
+    fixedLength: number;
+    /** The reference to the base type. */
+    type: ref.Type;
+
+    /**
+     * Accepts a Buffer instance that should be an already-populated with data
+     * for the ArrayType. The "length" of the Array is determined by searching
+     * through the buffer's contents until an aligned NULL pointer is encountered.
+     */
+    untilZeros(buffer: Buffer): {
+        [i: number]: T; length: number; toArray(): T[];
+        toJSON(): T[]; inspect(): string; buffer: Buffer; ref(): Buffer;
+    };
+
+    new (length?: number): {
```

Comment:
As discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes), you should probably just make `ArrayType` a class and this the constructor, even though the library itself doesn't explicitly use a class.